```
+ clang-tidy-18 --enable-check-profile --store-check-profile=clang-tidy-trace -quiet src/crafting_gui.cpp
72459 warnings generated.
Error: /home/runner/work/Cataclysm-DDA/Cataclysm-DDA/src/crafting_gui.cpp:1301:40: error: function 'select_crafter_and_crafting_recipe' exceeds recommended size/complexity thresholds [readability-function-size,-warnings-as-errors]
 1301 | std::pair<Character *, const recipe *> select_crafter_and_crafting_recipe( int &batch_size_out,
      |                                        ^
/home/runner/work/Cataclysm-DDA/Cataclysm-DDA/src/crafting_gui.cpp:1301:40: note: 845 statements (threshold 800)
Error: Process completed with exit code 123.
```
A _limit_ of _800_ lines in a method?  You've gotta be kidding me. 😆

--- a/src/crafting_gui.cpp
+++ b/src/crafting_gui.cpp
@@ -133,6 +133,8 @@ static int related_menu_fill( uilist &rmenu,
                               const recipe_subset &available );
 static item get_recipe_result_item( const recipe &rec, Character &crafter );
 static void compare_recipe_with_item( const item &recipe_item, Character &crafter );
+static void prioritize_components( const recipe &recipe, Character &crafter );
+static void deprioritize_components( const recipe &recipe );
 
 static std::string get_cat_unprefixed( std::string_view prefixed_name )
 {
@@ -2132,90 +2134,13 @@ std::pair<Character *, const recipe *> select_crafter_and_crafting_recipe( int &
             recalc_unread = highlight_unread_recipes;
             ui.invalidate_ui();
 
-            int new_filters_count = 0;
-            std::string added_filters;
-            const requirement_data &req = current[line]->simple_requirements();
-            const inventory &crafting_inv = crafter->crafting_inventory();
-            for( const std::vector<item_comp> &comp_list : req.get_components() ) {
-                for( const item_comp &i_comp : comp_list ) {
-                    std::string nname = item::nname( i_comp.type, 1 );
-                    int filter_pos = uistate.list_item_priority.find( nname );
-                    bool enough_materials = req.check_enough_materials(
-                                                i_comp, crafting_inv, current[line]->get_component_filter(), batch_size_out
-                                            );
-                    if( filter_pos == -1 && !enough_materials ) {
-                        new_filters_count++;
-                        added_filters += nname;
-                        added_filters += ",";
-                    }
-                }
-            }
-
-            if( new_filters_count > 0 ) {
-                if( !uistate.list_item_priority.empty()
-                    && !string_ends_with( uistate.list_item_priority, "," ) ) {
-                    uistate.list_item_priority += ",";
-                }
-                uistate.list_item_priority += added_filters;
-                uistate.list_item_priority_active = true;
-                std::vector<std::string> &hist = uistate.gethistory( "list_item_priority" );
-                if( hist.empty() || hist[hist.size() - 1] != uistate.list_item_priority ) {
-                    hist.push_back( uistate.list_item_priority );
-                }
-
-                popup( string_format( _( "Added %d components to the priority filter.\nAdded: %s\nNew Filter: %s" ),
-                                      new_filters_count, added_filters, uistate.list_item_priority ) );
-            } else {
-                popup( string_format( _( "Did not find anything to add to the priority filter.\nFilter: %s" ),
-                                      uistate.list_item_priority ) );
-            }
+            prioritize_components( *current[line], *crafter );
         } else if( action == "DEPRIORITIZE_COMPONENTS" ) {
             uistate.read_recipes.insert( current[line]->ident() );
             recalc_unread = highlight_unread_recipes;
             ui.invalidate_ui();
 
-            int removed_filters_count = 0;
-            std::string removed_filters;
-            const requirement_data &req = current[line]->simple_requirements();
-            for( const std::vector<item_comp> &comp_list : req.get_components() ) {
-                for( const item_comp &i_comp : comp_list ) {
-                    std::string nname = item::nname( i_comp.type, 1 );
-                    nname += ",";
-                    if( string_starts_with( uistate.list_item_priority, nname ) ) {
-                        removed_filters_count++;
-                        removed_filters += nname;
-                        uistate.list_item_priority.erase( 0, nname.length() );
-                    } else {
-                        std::string find_string = "," + nname;
-                        int filter_pos = uistate.list_item_priority.find( find_string );
-                        if( filter_pos > -1 ) {
-                            removed_filters_count++;
-                            removed_filters += nname;
-                            uistate.list_item_priority.replace( filter_pos, find_string.length(), "," );
-                        }
-                    }
-                }
-            }
-
-            if( !uistate.list_item_priority.empty() ) {
-                uistate.list_item_priority_active = true;
-
-                std::vector<std::string> &hist = uistate.gethistory( "list_item_priority" );
-                if( hist.empty() || hist[hist.size() - 1] != uistate.list_item_priority ) {
-                    hist.push_back( uistate.list_item_priority );
-                }
-            } else {
-                uistate.list_item_priority_active = false;
-            }
-
-            if( removed_filters_count > 0 ) {
-                popup( string_format(
-                           _( "Removed %d components from the priority filter.\nRemoved: %s\nNew Filter: %s" ),
-                           removed_filters_count, removed_filters, uistate.list_item_priority ) );
-            } else {
-                popup( string_format( _( "Did not find anything to remove from the priority filter.\nFilter: %s" ),
-                                      uistate.list_item_priority ) );
-            }
+            deprioritize_components( *current[line] );
         } else if( action == "HELP_KEYBINDINGS" ) {
             // Regenerate keybinding tips
             ui.mark_resize();
@@ -2437,6 +2362,93 @@ static void compare_recipe_with_item( const item &recipe_item, Character &crafte
         game_menus::inv::compare_item_menu menu( recipe_item, *to_compare );
         menu.show();
     } while( true );
+}
+
+static void prioritize_components( const recipe &recipe, Character &crafter )
+{
+    int new_filters_count = 0;
+    std::string added_filters;
+    const requirement_data &req = recipe.simple_requirements();
+    const inventory &crafting_inv = crafter.crafting_inventory();
+    for( const std::vector<item_comp> &comp_list : req.get_components() ) {
+        for( const item_comp &i_comp : comp_list ) {
+            std::string nname = item::nname( i_comp.type, 1 );
+            int filter_pos = uistate.list_item_priority.find( nname );
+            bool enough_materials = req.check_enough_materials(
+                    i_comp, crafting_inv, recipe.get_component_filter(), 1
+                    );
+            if( filter_pos == -1 && !enough_materials ) {
+                new_filters_count++;
+                added_filters += nname;
+                added_filters += ",";
+            }
+        }
+    }
+
+    if( new_filters_count > 0 ) {
+        if( !uistate.list_item_priority.empty()
+                && !string_ends_with( uistate.list_item_priority, "," ) ) {
+            uistate.list_item_priority += ",";
+        }
+        uistate.list_item_priority += added_filters;
+        uistate.list_item_priority_active = true;
+        std::vector<std::string> &hist = uistate.gethistory( "list_item_priority" );
+        if( hist.empty() || hist[hist.size() - 1] != uistate.list_item_priority ) {
+            hist.push_back( uistate.list_item_priority );
+        }
+
+        popup( string_format( _( "Added %d components to the priority filter.\nAdded: %s\nNew Filter: %s" ),
+                    new_filters_count, added_filters, uistate.list_item_priority ) );
+    } else {
+        popup( string_format( _( "Did not find anything to add to the priority filter.\n\nFilter: %s" ),
+                    uistate.list_item_priority ) );
+    }
+}
+
+static void deprioritize_components( const recipe &recipe )
+{
+    int removed_filters_count = 0;
+    std::string removed_filters;
+    const requirement_data &req = recipe.simple_requirements();
+    for( const std::vector<item_comp> &comp_list : req.get_components() ) {
+        for( const item_comp &i_comp : comp_list ) {
+            std::string nname = item::nname( i_comp.type, 1 );
+            nname += ",";
+            if( string_starts_with( uistate.list_item_priority, nname ) ) {
+                removed_filters_count++;
+                removed_filters += nname;
+                uistate.list_item_priority.erase( 0, nname.length() );
+            } else {
+                std::string find_string = "," + nname;
+                int filter_pos = uistate.list_item_priority.find( find_string );
+                if( filter_pos > -1 ) {
+                    removed_filters_count++;
+                    removed_filters += nname;
+                    uistate.list_item_priority.replace( filter_pos, find_string.length(), "," );
+                }
+            }
+        }
+    }
+
+    if( !uistate.list_item_priority.empty() ) {
+        uistate.list_item_priority_active = true;
+
+        std::vector<std::string> &hist = uistate.gethistory( "list_item_priority" );
+        if( hist.empty() || hist[hist.size() - 1] != uistate.list_item_priority ) {
+            hist.push_back( uistate.list_item_priority );
+        }
+    } else {
+        uistate.list_item_priority_active = false;
+    }
+
+    if( removed_filters_count > 0 ) {
+        popup( string_format(
+                   _( "Removed %d components from the priority filter.\nRemoved: %s\nNew Filter: %s" ),
+                   removed_filters_count, removed_filters, uistate.list_item_priority ) );
+    } else {
+        popup( string_format( _( "Did not find anything to remove from the priority filter.\nFilter: %s" ),
+                              uistate.list_item_priority ) );
+    }
 }
 
 static bool query_is_yes( std::string_view query )

--- a/src/crafting_gui.cpp
+++ b/src/crafting_gui.cpp
@@ -2375,8 +2375,8 @@ static void prioritize_components( const recipe &recipe, Character &crafter )
             std::string nname = item::nname( i_comp.type, 1 );
             int filter_pos = uistate.list_item_priority.find( nname );
             bool enough_materials = req.check_enough_materials(
-                    i_comp, crafting_inv, recipe.get_component_filter(), 1
-                    );
+                                        i_comp, crafting_inv, recipe.get_component_filter(), 1
+                                    );
             if( filter_pos == -1 && !enough_materials ) {
                 new_filters_count++;
                 added_filters += nname;
@@ -2387,7 +2387,7 @@ static void prioritize_components( const recipe &recipe, Character &crafter )
 
     if( new_filters_count > 0 ) {
         if( !uistate.list_item_priority.empty()
-                && !string_ends_with( uistate.list_item_priority, "," ) ) {
+            && !string_ends_with( uistate.list_item_priority, "," ) ) {
             uistate.list_item_priority += ",";
         }
         uistate.list_item_priority += added_filters;
@@ -2398,10 +2398,10 @@ static void prioritize_components( const recipe &recipe, Character &crafter )
         }
 
         popup( string_format( _( "Added %d components to the priority filter.\nAdded: %s\nNew Filter: %s" ),
-                    new_filters_count, added_filters, uistate.list_item_priority ) );
+                              new_filters_count, added_filters, uistate.list_item_priority ) );
     } else {
         popup( string_format( _( "Did not find anything to add to the priority filter.\n\nFilter: %s" ),
-                    uistate.list_item_priority ) );
+                              uistate.list_item_priority ) );
     }
 }
 

